src/commands/CommandUtils.ts

```diff
@@ -0,0 +1,36 @@
+import { REST } from '@discordjs/rest';
+import { Routes } from 'discord.js';
+import { CHAT_INPUT_COMMAND_INSTANCES } from '.';
+
+export default class CommandUtils {
+  static async registerCommands() {
+    if (process.env.BOT_TOKEN === undefined) {
```

Comment:
you can use the `envalid` library to validate this at start time. afterwards, you can encapsulate these values in a `config.ts` file that are typed to not be undefined

Reply:
will check that out!